### DISCUSSION TITLE
Script action based on temperature

### DISCUSSION THREAD

**Discussion by RikshaDriver** [gopro/labs#583] [Feature Requests]
Ask: The ask is to allow a camera temperature condition to facilitate scripting.

Use Case: I volunteer as a firefighter and would like to be able to script the following:

- If temperature is above X (Celsius) auto trigger recording start
- If temperature is above Y (Celsius) use lower resolution (i.e. 4K 24p 16:9), else use full resolution (i.e. 5.3k 24p 8:7)

**Comment by dnewman-gpsw** [comment]
While it might be doable, there aren't any surface temperature sensors, only temperatures of some components. So I'm how useful this would be.

**Reply by RikshaDriver** [reply]
I'm assuming the higher resolution thermal limiting relies on processor temps? This could be used in the script scenario to record at lower res & frame rates to reduce internal temperatures.

Ideally it would be a set and forget scenario when mounted to a helmet or chest as I wouldn't want to be fiddling around with settings between fighting bushfires.

This would be useful in other applications outside mine such as for on-person continuous recordings at the highest possible quality subject to the elements.


Are any of the internal components with temp sensors partially exposed, or towards the extremities of the camera if no ambient temp sensor exists to trigger auto record?
